src/compiler/checker.ts

```diff
@@ -10708,6 +10714,28 @@ namespace ts {
             return undefined;
         }
 
+        function getBindingElementNameText(element: BindingElement): string | undefined {
+            if (element.parent.kind === SyntaxKind.ObjectBindingPattern) {
+                const name = element.propertyName || element.name;
+                if (isComputedNonLiteralName(name as PropertyName)) return undefined;

```

Comment:
Consider moving this down to the analogous `case` clause to make it clearer that it's safe to assert `name.expression` as `LiteralExpression`